### ISSUE TITLE
Nullness : assertions in both branches do not refine the exit store

### ISSUE THREAD

**Issue by Ao-senXiong** [eisop/checker-framework#1231]
Take this code,
```
public class Demo{
    void test(){
        Set<Object> o = new HashSet<>();
        Object obj = null;
        if (true) {
            assert obj != null;
        } else {
            assert obj != null;
        }
        o.add(obj); // This should not raise an error
    }
}
```
The error is unexpected to me.
```
Demo.java:19: error: [argument.type.incompatible] incompatible argument for parameter arg0 of Set.add.
        o.add(obj); // This should not raise an error
              ^
  found   : @Nullable Object
  required: @NonNull Object
1 error
```

**Comment by Ao-senXiong** [comment]
To enable asseration to do flow refinement, use the [option](https://eisop.github.io/cf/manual/manual.html#type-refinement-assertions)
